app/domain/authentication/authn_k8s/application_identity.rb

```diff
@@ -45,11 +45,25 @@ def constraints
       end
 
       def container_name
-        annotation_name = "authentication-container-name"
-        annotation_value("authn-k8s/#{@service_id}/#{annotation_name}") ||
-          annotation_value("authn-k8s/#{annotation_name}") ||
-          annotation_value("kubernetes/#{annotation_name}") ||
-          "authenticator"
+        @container_name ||= annotation_value("authn-k8s/#{@service_id}/#{authentication_container_name}") ||
+          annotation_value("authn-k8s/#{authentication_container_name}") ||
+          annotation_value("kubernetes/#{authentication_container_name}") ||
+          default_authentication_container_name
+      end
+
+      def authentication_container_name
```

Comment:
This can also be a [class constant](https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/constants.html) instead of a method that just returns a literal.